docs/_posts/examples/3400-01-15-animate-a-line.html

```diff
@@ -0,0 +1,126 @@
+---
+layout: example
+category: example
+title: Animate a line
+description: Animate a line by updating a GeoJSON source on each frame.
+tags:
+- layers
+- sources
+---
+<style>
+button {
+    position: absolute;
+    margin: 20px;
+}
+
+#pause::after {
+    content: 'Pause';
+}
+
+#pause.pause::after {
+    content: 'Play';
+}
+</style>
+<div id='map'></div>
+<button id='pause'></button>
+<script>
+var map = new mapboxgl.Map({
+    container: 'map',
+    style: 'mapbox://styles/mapbox/streets-v8',
+    center: [0, 0],
+    zoom: 0.5
+});
+
+// Create a GeoJSON source with an empty lineString.
+var geojson = {
+    "type": "FeatureCollection",
+    "features": [{
+        "type": "Feature",
+        "geometry": {
+            "type": "LineString",
+            "coordinates": [
+                [0, 0]
+            ]
+        }
+    }]
+};
+
+var speedFactor = 30; // number of frames per longitude degree
+var animation;
+var startTime = 0;
+var progress = 0; // progress = timestamp - startTime
+var pause = false;
```

Comment:
@mollymerp this is used to flag the pause status set by either window focus or the pause button. I think 'pauseAnimation' may be more accurate.